src/android/toga_android/app.py

```diff
@@ -39,6 +44,21 @@ def onDestroy(self):
     def onRestart(self):
         print("Toga app: onRestart")
 
+    def onActivityResult(self, requestCode, resultCode, resultData):
+        """
+        Callback method, called from MainActivity when an Intent ends
+
+        :param int requestCode: The integer request code originally supplied to startActivityForResult(),
+                                allowing you to identify who this result came from.
+        :param int resultCode: The integer result code returned by the child activity through its setResult().
+        :param Intent resultData: An Intent, which can return result data to the caller (various data can be attached
+                                  to Intent "extras").
+        """
+        print("Toga app: onActivityResult")
```

Comment:
> preference would be to do this conversion in the Android backend and return a path that is compatible with standard Python file methods

Well, I have thought about this as well and it could be done pretty easily.
But then, we would need to reject all files selected from content providers except for the local storage content provider.
So no Cloud, no image provider, maybe not even downloads provider.

In addition there is the problem that in Android 11 the file I/O will not be supported anymore on external storage:
https://developer.android.com/about/versions/11/privacy/storage
So, using file paths on Android should be considered deprecated already.

That's why I think we should stick to the content URI params and result of open_file_dialog.
I also mentioned in the PR description that the user will need to use SAF method to actually read and write files.
This can be achieved by calling them through app._impl.whatever but I really don't like this idea!
I therefore suggest that in Toga, we should use abstract "file identifier strings" to reference files.
On Android, they would be content URIs, on all other platforms, they would be file paths.
We could then add some high-level cross-platform file-util module for accessing files.
The module would accept the "file identifier strings" and support following methods:
read_text_file (with handling encoding and EOL conversion)
read_binary_file
write_text_file (with handling encoding and EOL conversion)
delete_file
create_folder
delete_folder

Just my 2 cents...

Reply:
Hrm. That's... annoying... but equally annoying is that the reasons that make a lot of sense. :-)

I agree that a high level "just get me the file content" API would be extremely useful. The general problem of "download this file in the background" is a universal one; it's just a little more obvious on mobile (and Android is clearly forcing the issue by locking down the security on those APIs). However, this does give us an opportunity to provide an API that is Pathlib/File duck-compatible, but abstracts network (or file) access.

We don't need to solve that problem right now - but we do need to work out how to *not* box ourselves into a corner for a future where this API exists.

I'm not wild about the idea of Android returning a different type of result than other platforms; however, I'm also not completely sure I have a better idea as an alternative. I guess it's easy enough to distinguish `content://` and `file://` from `/home/myname` and `c:\Users`...

Reply:
OK, I added some method documentation in toga\window.py open_file_dialog and select_folder_dialog to make it clear to the user that on Android, he should work with content URIs